Algorithm.CSharp/OptionChainProviderAlgorithm.cs

```diff
@@ -37,19 +37,25 @@ namespace QuantConnect.Algorithm.CSharp
     public class OptionChainProviderAlgorithm : QCAlgorithm
     {
         private Symbol _equitySymbol;
-     
+        private Symbol _optionContract = string.Empty;
+
         public override void Initialize()
         {
-            SetStartDate(2017, 06, 01);
-            SetEndDate(2017, 07, 01);
+            SetStartDate(2015, 12, 24);
+            SetEndDate(2015, 12, 24);
             SetCash(100000);
-            var equity = AddEquity("AMZN", Resolution.Minute);
+            var equity = AddEquity("GOOG", Resolution.Minute);
             _equitySymbol = equity.Symbol;
         }
 
         public override void OnData(Slice data)
         {
-            if (!Portfolio.Invested)
+            if (!Portfolio[_equitySymbol].Invested)
+            {
+                MarketOrder(_equitySymbol, 100);
+            }
+
+            if (!(Securities.ContainsKey(_optionContract) && Portfolio[_optionContract].Invested))
```

Comment:
do we need this invested condition here? This block of code is only adding the contract